lib/rucio/core/identity.py

```diff
@@ -47,7 +49,12 @@ def add_identity(identity, type, email, password=None, session=None):
 
     if type == IdentityType.USERPASS and password is not None:
         salt = os.urandom(255)  # make sure the salt has the length of the hash
-        password = hashlib.sha256('%s%s' % (salt, password.encode('utf-8'))).hexdigest()  # hash it
+        if six.PY3:
+            salt = salt.decode()
+            salted_password = ('%s%s' % (salt, password)).encode()
+        else:
+            salted_password = '%s%s' % (salt, str(password))
```

Comment:
is it not necessary to do the encode('utf-8') in py2 anymore like previously?

Reply:
I have added the encoding for py3 compatibility but at the wrong place, because the whole string needs to be encoded, not the password. But back then, my wrong fix was enough to let the pylint python3 test pass.